lib/spack/spack/fetch_strategy.py

```diff
@@ -520,7 +526,16 @@ def archive(self, destination, **kwargs):
                 tar.add_default_arg('--exclude=%s' % p)
 
         with working_dir(self.stage.path):
-            tar('-czf', destination, os.path.basename(self.stage.source_path))
+            if self.stage.srcdir:
```

Comment:
Is this change addressing potential naming conflicts?  Or ensuring any surviving archive has a meaningful name?

Reply:
The latter: `srcdir` stores whatever meaningful name can be extracted (e.g. the basename of the URL). If one cannot be extracted, then we resort to creating a tarball that expands to `spack-src/`.